Formula/r/recc.rb

```diff
@@ -0,0 +1,180 @@
+class Recc < Formula
+  desc "Remote Execution Caching Compiler"
+  homepage "https://buildgrid.gitlab.io/recc"
+  url "https://gitlab.com/BuildGrid/buildbox/buildbox/-/archive/1.2.21/buildbox-1.2.21.tar.gz"
+  sha256 "7ac8e96ae9c614659b9f16a988a89609bf9c0c5413241f9912f2d0d330100997"
+  license "Apache-2.0"
+  head "https://gitlab.com/BuildGrid/buildbox/buildbox.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "tomlplusplus" => :build
+  depends_on "abseil"
+  depends_on "c-ares"
+  depends_on "glog"
+  depends_on "grpc"
+  depends_on "openssl@3"
+  depends_on "protobuf"
+  depends_on "re2"
+  uses_from_macos "zlib"
+
+  on_macos do
+    depends_on "gflags"
+  end
+
+  on_linux do
+    depends_on "pkg-config" => :build
+    depends_on "util-linux"
+  end
+
+  def install
+    buildbox_cmake_args = %W[
+      -DCASD=ON
+      -DCASD_BUILD_BENCHMARK=OFF
+      -DCASDOWNLOAD=OFF
+      -DCASUPLOAD=OFF
+      -DFUSE=OFF
+      -DLOGSTREAMRECEIVER=OFF
+      -DLOGSTREAMTAIL=OFF
+      -DOUTPUTSTREAMER=OFF
+      -DRECC=ON
+      -DREXPLORER=OFF
+      -DRUMBA=OFF
+      -DRUN_BUBBLEWRAP=OFF
+      -DRUN_HOSTTOOLS=ON
+      -DRUN_OCI=OFF
+      -DRUN_USERCHROOT=OFF
+      -DTREXE=OFF
+      -DWORKER=OFF
+      -DRECC_CONFIG_PREFIX_DIR=#{etc}
+    ]
+
+    system "cmake", "-S", ".", "-B", "build", *buildbox_cmake_args, *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    %w[cc c++ gcc g++ clang clang++].each do |compiler|
+      (bin/"recc-#{compiler}").write <<~EOS
+        #!/usr/bin/env sh
+
+        #{bin}/recc $(command -v #{compiler}) "$@"
```

Comment:
Is there any scope to upstream some/all of these helpers scripts?

Reply:
It makes sense to add the `recc-${compiler}` script. I'll also revisit our previous discussion on defining some default values for those included in `recc.conf`.

If there are no major concerns, would it be acceptable to merge the formula in the meantime?

Reply:
> If there are no major concerns, would it be acceptable to merge the formula in the meantime?

I'm afraid this is a fairly major concern to me given how much actual runtime code (not just building the software) is being shipped in this formula. It's the bulk of this formula file and this is unusual in Homebrew.

What's your relationship with upstream? Are you a user/contributor/maintainer?

Would it be tenable to ship a first version of this formulae with no compiler/server helper scripts and an upstream configuration file instead?

Reply:
I am one of the upstream maintainers.  I would prefer shipping it in out-of-the box working state. I will move the PR to draft state and come back after adding config and helper scripts upstream.

Reply:
Thanks! Appreciate your patience and hard work here ❤️

Reply:
As discussed, I added the wrappers in upstream project and updated the formula to use them